File: ColdDream.UniApp/src/api/banner.ts
import { request } from '@/utils/request';

export interface Banner {
    id: string;
    imageUrl: string;
    title: string;
    tag: string;
    targetUrl: string;
    sortOrder: number;
}

export const getBanners = () => {
    return request<Banner[]>({
        url: '/banner',
        method: 'GET',
    });
};

File: ColdDream.Api/Models/Banner.cs
using System.ComponentModel.DataAnnotations;

namespace ColdDream.Api.Models;

public class Banner
{
    public Guid Id { get; set; }
    
    [Required]
    public string ImageUrl { get; set; } = string.Empty;
    
    public string Title { get; set; } = string.Empty;
    
    public string Tag { get; set; } = string.Empty;
    
    public string TargetUrl { get; set; } = string.Empty;
    
    public int SortOrder { get; set; }
    
    public bool IsActive { get; set; } = true;
    
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}
